notebook/static/tree/js/notebooklist.js

```diff
@@ -697,6 +707,71 @@ define([
         });
     };
 
+    NotebookList.prototype.move_selected = function() {
+        var that = this;
+
+        // TODO(nhdaly): Support moving multiple items at once.
+        if (that.selected.length !== 1){
+            return;
+        }
+
+        var item_path = that.selected[0].path;
+        var item_name = that.selected[0].name;
+        var item_type = that.selected[0].type;
+
+        // Open a dialog to enter the new path, with current path as default.
+        var input = $('<input/>').attr('type','text').attr('size','25').addClass('form-control')
+            .val(utils.url_path_join('/', that.notebook_path));
+        var dialog_body = $('<div/>').append(
+            $("<p/>").addClass("rename-message")
+                .text('Enter new destination directory path for '+ item_type + ':')
+        ).append(
+            $("<br/>")
+        ).append(input);
+        var d = dialog.modal({
+            title : "Move "+ item_type,
+            body : dialog_body,
+            buttons : {
+                Cancel : {},
+                OK : {
+                    class: "btn-primary",
+                    click: function() {
+                      // Construct the new path using the user input and its name.
+                        var new_path = utils.url_path_join(input.val(), item_name)
+                        that.contents.rename(item_path, new_path).then(function() {
+                            that.load_list();
+                        }).catch(function(e) { 
+                            dialog.modal({
+                                title: "Move Failed",
+                                body: $('<div/>')
+                                    .text("An error occurred while moving \"" + item_name + "\" from \"" + item_path + "\" to \"" + new_path + "\".")
+                                    .append($('<div/>')
+                                        .addClass('alert alert-danger')
+                                        .text(e.message || e)),
+                                buttons: {
+                                    OK: {'class': 'btn-primary'}
+                                }
+                            });
+                            console.warn('Error durring moving :', e);
+                        });
+                    }
+                }
+            },
+            open : function () {
+                // Upon ENTER, click the OK button.
+                input.keydown(function (event) {
+                    if (event.which === keyboard.keycodes.enter) {
+                        d.find('.btn-primary').first().click();
+                        return false;
+                    }
+                });
+                input.focus();
+                // Highlight the current path in the input box.
+                input.select();
```

Comment:
@captainsafia: I believe the flow you have described is exactly how the `Rename` button already works. You can use `Rename` to move a file to a different directory by changing the relative path. E.g:
To move a file to a parent directory:
 1. Select checkbox on file.
 2. Press Rename. The textbox is pre-populated with just the filename + extension (e.g. "my_notebook.pynb")
 3. Prepend `../` to the name (e.g. changing it to "../my_notebook.pynb")
 4. Enter.

That's all done with relative paths. Is that what you were describing? However, I don't think this is intuitive for a user not familiar with Unix, who wouldn't think to rename a file to "../<filename>" to move it up a directory. That's why I wanted to add a Move button.


@ellisonbg + @captainsafia: Yeah, wow, that *is* a slick UI. I love it! That said, I think it would be good to separate that fancy UI into a separate PR? It works almost the same as this code, but builds a fancier UI on top of it. If we're happy with the semantics in this PR, I'd prefer starting a new one for the fancier interface. Does that sound okay?